File: src/main/java/com/vaadin/demo/component/icons/IconsColor.java
package com.vaadin.demo.component.icons;

import com.vaadin.demo.DemoExporter;
import com.vaadin.flow.component.html.Div;
import com.vaadin.flow.component.icon.FontIcon;
import com.vaadin.flow.component.icon.SvgIcon;
import com.vaadin.flow.component.orderedlayout.FlexComponent;
import com.vaadin.flow.component.orderedlayout.HorizontalLayout;
import com.vaadin.flow.router.Route;

@Route("icons-color")
public class IconsColor extends Div {

    public IconsColor() {
        HorizontalLayout layout = new HorizontalLayout();
        layout.setSpacing(true);
        layout.setAlignItems(FlexComponent.Alignment.CENTER);

        // tag::snippet[]
        SvgIcon svgIcon = new SvgIcon("/icons/code-branch.svg", "svg-branch.svg");
        svgIcon.setColor("red");

        FontIcon fontIcon = new FontIcon("fa", "fa-user");
        fontIcon.setColor("red");

        layout.add(svgIcon, fontIcon);
        // end::snippet[]
        add(layout);
    }

    public static class Exporter extends DemoExporter<IconsColor> { // hidden-source-line
    } // hidden-source-line
}

File: src/main/frontend/demo/component/icons/icons-color.ts
import 'Frontend/demo/init'; // hidden-source-line
import '@vaadin/horizontal-layout';
import '@vaadin/icon';
import { html, LitElement } from 'lit';
import { customElement } from 'lit/decorators.js';
import { applyTheme } from 'Frontend/generated/theme';
import codeBranch from '../../../../src/main/resources/icons/code-branch.svg?url';

@customElement('icons-color')
export class Example extends LitElement {
  protected override createRenderRoot() {
    const root = super.createRenderRoot();
    // Apply custom theme (only supported if your app uses one)
    applyTheme(root);
    return root;
  }

  protected override render() {
    return html`
      <vaadin-horizontal-layout theme="spacing" class="items-center">
        <!-- tag::snippet[] -->
        <vaadin-icon src="${codeBranch}" style="color: red"></vaadin-icon>
        <vaadin-icon icon-class="fa fa-user" style="color: red"></vaadin-icon>
        <!-- end::snippet[] -->
      </vaadin-horizontal-layout>
    `;
  }
}
